src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,64 @@
 package mate.academy;
 
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    public static final int THREAD_POOL_MIN_SIZE = 1;
+    public static final int SHUTDOWN_TIMEOUT_MINUTES = 2;
+    private final ThreadPoolExecutor executor;
+    private final List<EventListener> listeners;
 
+    public EventManager() {
+        this.executor = new ThreadPoolExecutor(
+                THREAD_POOL_MIN_SIZE, THREAD_POOL_MIN_SIZE,
+                0L, TimeUnit.MILLISECONDS,
+                new LinkedBlockingQueue<>());
+        this.listeners = new CopyOnWriteArrayList<>();
     }
 
-    public void deregisterListener(EventListener listener) {
+    public void registerListener(EventListener listener) {
+        listeners.add(listener);
+        adjustThreadPoolSize();
+    }
 
+    public void deregisterListener(EventListener listener) {
+        listeners.remove(listener);
+        adjustThreadPoolSize();
     }
 
     public void notifyEvent(Event event) {
-
+        Iterator<EventListener> iterator = listeners.iterator();
+        while (iterator.hasNext()) {
```

Comment:
:+1: 
But we have to remember that once `notifyEvent` is called - other new listeners that are added in parallel will be ignored

Reply:
It was ai-mentor's idea. We will notify only those listeners, who was subscribed at calling moment